File: client/src/classes/EventDay.ts
export default class EventDay {
  id?: string;
  date = new Date(0);
  description = '';
  eventId? = '';

  constructor(i?: Event) {
    PF.H.Classes.BuildClass(this, i);
  }

  static Create(eventId: string, date = new Date()): EventDay {
    const item = new EventDay();
    item.id = PF.H.Id.UUID();
    item.eventId = eventId;
    item.date = date;
    return item;
  }
}

File: server/models/EventDay.go
package models

import (
	"time"

	"github.com/google/uuid"
	"github.com/uptrace/bun"
)

type EventDay struct {
	bun.BaseModel `bun:"event_days,alias:event_days"`
	ID            uuid.NullUUID `bun:"id,pk,type:uuid,default:uuid_generate_v4()" json:"id"`
	Date          time.Time     `bun:"item_date" json:"date"`
	Description   string        `json:"description"`

	EventID uuid.NullUUID `json:"eventId"`
}

type EventDays []*EventDay

type EventDaysWithCount struct {
	EventDays EventDays `json:"items"`
	Count     int       `json:"count"`
}
